### PR TITLE
Next convert profile props

### PR DESCRIPTION
**Pull request by gejohnston**
This PR adds logic to the `zowe config convert-profiles` command to also convert old V1 profile property names to their V2 compliant profile property names.

The following table shows the property name conversions.

| Old V1 profile property name | New V2 profile property name |
| ---------------------------- | ---------------------------- |
| hostname                     | host                         |
| username                     | user                         |
| pass                         | password                     |

### CODE REVIEW

File: __tests__/__integration__/imperative/__tests__/__integration__/cli/config/__resources__/profiles_with_v1_names/v1profile/myv1profile.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+hostname: 'convert hostname to host'
+username: 'convert username to user'
+pass: 'convert pass to password'
+token: 'convert token to takenValue'
```
Comment (t1m0thyj) [inline __tests__/__integration__/imperative/__tests__/__integration__/cli/config/__resources__/profiles_with_v1_names/v1profile/myv1profile.yaml:4]: Are we sure that converting `token` to `tokenValue` is always desirable?

For some plugins like Endevor Bridge for Git, "token" is used for a different purpose than connecting to the mainframe. Thanks to @awharn for pointing this out 🙂

According to the web help for the [`zowe endevor-bridge-for-git endevor-connection create`](https://docs.zowe.org/stable/web_help/index.html?p=zowe_endevor-bridge-for-git_endevor-connection_create) command:
image

Comment (gejohnston) [reply __tests__/__integration__/imperative/__tests__/__integration__/cli/config/__resources__/profiles_with_v1_names/v1profile/myv1profile.yaml:4]: The conversion of token came from this item in the story:

    Info from Bridge for Git
     - token -> tokenValue should be included
 
We can revisit this question in a standup meeting to confirm if this is the desired action.

**Review by zFernand0 — COMMENTED** [review]
Implementation LGTM as it is.
waiting on the `token -> tokenValue` discussion to approve😋

File: packages/config/__tests__/ConfigBuilder.test.ts [hunk 2]
```diff
@@ -357,4 +357,42 @@ describe("Config Builder tests", () => {
...
+            profiles: {
+                zosmf_LPAR1: {
+                    type: "zosmf",
+                    properties: {
+                        host: "should change to host",
+                        user: "should change to user",
+                        password: "area51",
+                        tokenValue: "area51"
+                    },
+                    secure: ["password", "tokenValue"]
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/config/__tests__/ConfigBuilder.test.ts:387]: If I understand your question, the answer is yes. Let me explain, and you can determine if that answers your question.

At this point in time, the config within the convertResult object is an in-memory version of the config. ConfigBuilder.convert() returns this convertResult to the ConvertProfilesHandler, which then saves the config. It is the Config.save() function that removes the secure property names and values from the "properties" portion of the config that is to be stored on disk, and save() also stores the values of those secure properties into the secure credential store.

File: packages/config/src/ConfigBuilder.ts [hunk 3]
```diff
@@ -136,10 +136,75 @@ export class ConfigBuilder {
...
+     * @param conversionResult The conversion result structure in which we shall
+     *      rename obsolete property names to their V2-compliant names.
+     */
+    private static convertPropNames(conversionResult: IConfigConvertResult): void {
+        const nameConversions = [
+            ["hostname", "host"],
+            ["username", "user"],
+            ["pass", "password"],
+            ["token", "tokenValue"]
+        ];
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/config/src/ConfigBuilder.ts:158]: Or to remove them if we decide that we should not be converting "token".

File: packages/config/__tests__/ConfigBuilder.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zFernand0) [reply packages/config/__tests__/ConfigBuilder.test.ts]: thank you! : )
this definitely clarifies things for me! 😋

**Review by zFernand0 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent Changes
+
+- Enhancement: Convert previously used profile property names into V2-compliant property names during the `zowe config convert-profiles` command. Conversions are: hostname -> host, username -> user, pass -> password, token -> tokenValue.
```
Comment (t1m0thyj) [inline CHANGELOG.md:7]: `token -> tokenValue` no longer applies here.

**Review by t1m0thyj — APPROVED** [review]
LGTM, just needs a changelog update 🙂

**Pull request merged**